.github/workflows/main.yaml

```diff
@@ -0,0 +1,164 @@
+        if: inputs.self-build!=1
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.build-artifact-name }}
+          path: ${{ inputs.build-artifact-path }}
+      - uses: actions/setup-python@v3
+        with:
+          python-version: 3.10.2
+      - name: Install node
+        uses: actions/setup-node@v3
+        if: ${{ contains(inputs.browser, 'node') || inputs.runner == 'playwright' }}
+        with:
+          node-version: ${{ inputs.driver-version }}
+
+      - name: Cache Playwright browsers
+        uses: actions/cache@v3
+        if: ${{ inputs.runner == 'playwright' }}
+        with:
+          path: .cache/ms-playwright
+          key: ${{ runner.os }}-playwright-${{ inputs.browser-version }}
+
+      - name: Install playwright
+        shell: bash -l {0}
+        if: ${{ inputs.runner == 'playwright' }}
+        run: |
+          python3 -m pip install playwright==${{ inputs.browser-version }}
+          # TODO: install only browsers that are required
+          python3 -m playwright install --with-deps
+
+      - name: Install firefox
+        uses: browser-actions/setup-firefox@latest
+        if: ${{ inputs.runner == 'selenium' && contains(inputs.browser, 'firefox') }}
+        with:
+          firefox-version: ${{ inputs.browser-version }}
+
+      - name: Install geckodriver
+        uses: browser-actions/setup-geckodriver@latest
+        if: ${{ inputs.runner == 'selenium' && contains(inputs.browser, 'firefox') }}
+        with:
+          geckodriver-version: ${{ inputs.driver-version }}
+
+      - name: Install chrome
+        uses: browser-actions/setup-chrome@latest
+        if: ${{ inputs.runner == 'selenium' && contains(inputs.browser, 'chrome') }}
+        with:
+          chrome-version: ${{ inputs.browser-version }}
+
+      - name: Install chromedriver
+        if: ${{ inputs.runner == 'selenium' && contains(inputs.browser, 'chrome') }}
+        uses: nanasess/setup-chromedriver@v1
+
+      - name: Enable Safari Driver
+        if: ${{ inputs.runner == 'selenium' && contains(inputs.browser, 'safari') && contains(runner.os, 'macos') }}
+        run: |
+          sudo safaridriver --enable
+          # Only one Safari browser instance can be active at any given time
+          echo "STANDALONE_REFRESH=${{ inputs.refresh }}" >> $GITHUB_ENV
+
+      - name: Install requirements
+        shell: bash -l {0}
+        if: inputs.self-build==1
+        run: |
+          python3 -m pip install -e .
+          python3 -m pip install pytest-cov
+          # Currently we only install the package for dependencies.
+          # We then uninstall it otherwise tests fails due to pytest hook being
+          # registered twice.
+          python3 -m pip uninstall -y pytest-pyodide
+          which npm && npm install -g npm && npm update
+          which npm && npm install node-fetch@2
+      - name: Install pytest-pyodide for workflow
+        shell: bash -l {0}
+        if: inputs.self-build!=1
```

Comment:
It would be nice if we can remove this `self-build` condition later. (No action required for this PR).